src/Microsoft.AspNetCore.Http.Abstractions/Extensions/EndpointModel.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder
```

Comment:
Do we know the reason why these need to be in the HttpAbstractions? I think it's for the following reasons:
1. So that middleware can use the builder to become routerware
2. So that cross-cutting concerns like auth don't have to depend on routing

Is this valuable? I'm not sure that it matters what repo we define this stuff in.